### PR TITLE
[cmake] Build iree-sample-deps with iree-test-deps

### PR DESCRIPTION
**Pull request by kuhar**
This is so that running all tests with `ctest` does not require building both `iree-test-deps` and `iree-sample-deps`. At the same time, it is still possible to build samples without building all the other test dependencies.

Update all build scripts so that they build `iree-test-deps` only.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -748,9 +748,23 @@ if(IREE_BUILD_DOCS)
...
+)
+
 # Testing rules that require generation will add dependencies to this target.
 # This allows them to be EXCLUDE_FROM_ALL but still invokable.
-add_custom_target(iree-test-deps COMMENT "Building IREE test deps")
+add_custom_target(iree-test-deps
+  COMMENT
+    "Building IREE test deps"
+  DEPENDS
+    iree-sample-deps
```
Comment (ScottTodd) [inline CMakeLists.txt:766]: Ah, this is cleaner than what https://github.com/iree-org/iree/pull/11287 had originally
image
Nice.

File: CMakeLists.txt [hunk 2]
```diff
@@ -748,9 +748,23 @@ if(IREE_BUILD_DOCS)
   add_custom_target(iree-doc ALL)
 endif()
 
+# Samples may require additional files to be built/configured and will add
+# dependencies to this target.
+# Note: These will be automatically built with test dependencies
+# (`iree-test-deps`).
+add_custom_target(iree-sample-deps
```
Comment (ScottTodd) [inline CMakeLists.txt:755]: Can you also remove the explicit builds from our scripts? There are several with this pattern: https://github.com/iree-org/iree/blob/eadd7498d7cbe4564ab5f12321dfeb882f7ff1d7/build_tools/cmake/build_all.sh#L62-L68

That can just be
```cmake
echo "Building test deps"
echo "------------------"
"$CMAKE_BIN" --build "${BUILD_DIR}" --target iree-test-deps -- -k 0
```
with this change

Comment (kuhar) [reply CMakeLists.txt:755]: I wanted to do that in a follow-up PR. Do you think it's better to do it in one go?

Comment (ScottTodd) [reply CMakeLists.txt:755]: Yeah, I don't see a reason to split that into two PRs.

Comment (ScottTodd) [reply CMakeLists.txt:755]: Right now the build output looks like this:
```
+ echo 'Building test deps'
Building test deps
------------------
+ echo ------------------
+ /usr/bin/cmake --build full-build-dir --target iree-test-deps -- -k 0
[0/2] Re-checking globbed directories...
[1/827] Generating e2e_matmul_mmt4d_f32_large_llvm-cpu_local-task.mlir, e2e_matmul_mmt4d_f32_large_llvm-cpu_local-task.yaml

...
[750/827] Generating simple_mul.spv
[751/827] Generating simple_mul_inplace.spv
...

+ echo 'Building sample deps'
+ echo ------------------
+ /usr/bin/cmake --build full-build-dir --target iree-sample-deps -- -k 0
Building sample deps
------------------
[0/2] Re-checking globbed directories...
ninja: no work to do.
```
(https://github.com/iree-org/iree/actions/runs/4118092232/jobs/7110194596#step:4:7501)

**Pull request merged**